### ISSUE TITLE
nosetests error on ubuntu with kernal 3.4, 3.5

### ISSUE THREAD

**Issue by yungyuc** [solvcon/pastsolvcon#58]
Originally reported by: **BitBucket: [Robertlin](http://bitbucket.org/Robertlin), GitHub: [Robertlin](http://github.com/Robertlin)**

---

proc stat format is different between ubuntu and solvcon default.
for solvcon it's 44 items and for ubuntu it 52 item
So index out of range will occur in the for loop (for more detail see below error log of nosetests)

the attached file is proc stat file of ubuntu.

error log of nosetests:
## ERROR: test_runtimestat (solvcon.tests.test_anchor.TestAnchor)

Traceback (most recent call last):
  File "/home/robertlin/solvcon/solvcon/solvcon/tests/test_anchor.py", line 96, in test_runtimestat
    svr.march(self.time, self.time_increment, self.nsteps)
  File "/home/robertlin/solvcon/solvcon/solvcon/solver.py", line 335, in march
    self.runanchors('postfull')
  File "/home/robertlin/solvcon/solvcon/solvcon/anchor.py", line 104, in **call**
    func()
  File "/home/robertlin/solvcon/solvcon/solvcon/anchor.py", line 473, in postfull
    rec = self._get_record()
  File "/home/robertlin/solvcon/solvcon/solvcon/anchor.py", line 350, in _get_record
    pstat = self.get_pstat()
  File "/home/robertlin/solvcon/solvcon/solvcon/anchor.py", line 307, in get_pstat
    key, typ = cls.PSTAT_KEYS[it]
IndexError: list index out of range

---
- Bitbucket: https://bitbucket.org/solvcon/solvcon/issue/58

**Comment by yungyuc** [comment]
_Original comment by_ **Robert Lin (BitBucket: [gotwlin](http://bitbucket.org/gotwlin), GitHub: [gotwlin](http://github.com/gotwlin))**:

---

update 2013/01/26
since solvcon update to separate the data by split()
we may treat

```
    seq_printf(m, " 0 0 0 0 0 0 0");
```

as 7 depend unit. 
with
        seq_put_decimal_ll(m, ' ', 0);
so in all version that should be 52 items.

## 

try to get linux kernel from www.kernel.org.

depends on function 

```
#!c
static int do_task_stat(struct seq_file *m, struct pid_namespace *ns,
            struct pid *pid, struct task_struct *task, int whole)
{.....
    seq_printf(m, "%d (%s) %c", pid_nr_ns(pid, ns), tcomm, state);
    seq_put_decimal_ll(m, ' ', ppid);
    seq_put_decimal_ll(m, ' ', pgid);
    seq_put_decimal_ll(m, ' ', sid);
    seq_put_decimal_ll(m, ' ', tty_nr);
    seq_put_decimal_ll(m, ' ', tty_pgrp);
    seq_put_decimal_ull(m, ' ', task->flags);
    seq_put_decimal_ull(m, ' ', min_flt);
    seq_put_decimal_ull(m, ' ', cmin_flt);
    seq_put_decimal_ull(m, ' ', maj_flt);
    seq_put_decimal_ull(m, ' ', cmaj_flt);
    seq_put_decimal_ull(m, ' ', cputime_to_clock_t(utime));
    seq_put_decimal_ull(m, ' ', cputime_to_clock_t(stime));
    seq_put_decimal_ll(m, ' ', cputime_to_clock_t(cutime));
    seq_put_decimal_ll(m, ' ', cputime_to_clock_t(cstime));
    seq_put_decimal_ll(m, ' ', priority);
    seq_put_decimal_ll(m, ' ', nice);
    seq_put_decimal_ll(m, ' ', num_threads);
    seq_put_decimal_ull(m, ' ', 0);
    seq_put_decimal_ull(m, ' ', start_time);
    seq_put_decimal_ull(m, ' ', vsize);
    seq_put_decimal_ull(m, ' ', mm ? get_mm_rss(mm) : 0);
    seq_put_decimal_ull(m, ' ', rsslim);
    seq_put_decimal_ull(m, ' ', mm ? (permitted ? mm->start_code : 1) : 0);
    seq_put_decimal_ull(m, ' ', mm ? (permitted ? mm->end_code : 1) : 0);
    seq_put_decimal_ull(m, ' ', (permitted && mm) ? mm->start_stack : 0);
    seq_put_decimal_ull(m, ' ', esp);
    seq_put_decimal_ull(m, ' ', eip);
    /* The signal information here is obsolete.
     * It must be decimal for Linux 2.0 compatibility.
     * Use /proc/#/status for real-time signals.
     */
    seq_put_decimal_ull(m, ' ', task->pending.signal.sig[0] & 0x7fffffffUL);
    seq_put_decimal_ull(m, ' ', task->blocked.sig[0] & 0x7fffffffUL);
    seq_put_decimal_ull(m, ' ', sigign.sig[0] & 0x7fffffffUL);
    seq_put_decimal_ull(m, ' ', sigcatch.sig[0] & 0x7fffffffUL);
    seq_put_decimal_ull(m, ' ', wchan);
    seq_put_decimal_ull(m, ' ', 0);
    seq_put_decimal_ull(m, ' ', 0);
    seq_put_decimal_ll(m, ' ', task->exit_signal);
    seq_put_decimal_ll(m, ' ', task_cpu(task));
    seq_put_decimal_ull(m, ' ', task->rt_priority);
    seq_put_decimal_ull(m, ' ', task->policy);
    seq_put_decimal_ull(m, ' ', delayacct_blkio_ticks(task));
    seq_put_decimal_ull(m, ' ', cputime_to_clock_t(gtime));
    seq_put_decimal_ll(m, ' ', cputime_to_clock_t(cgtime));

    if (mm && permitted) {
        seq_put_decimal_ull(m, ' ', mm->start_data);
        seq_put_decimal_ull(m, ' ', mm->end_data);
        seq_put_decimal_ull(m, ' ', mm->start_brk);
        seq_put_decimal_ull(m, ' ', mm->arg_start);
        seq_put_decimal_ull(m, ' ', mm->arg_end);
        seq_put_decimal_ull(m, ' ', mm->env_start);
        seq_put_decimal_ull(m, ' ', mm->env_end);
    } else
        seq_printf(m, " 0 0 0 0 0 0 0");

    if (permitted)
        seq_put_decimal_ll(m, ' ', task->exit_code);
    else
        seq_put_decimal_ll(m, ' ', 0);

```

we can tell that ubuntu may define the permitted. to print extra lines.

below

```
#!c
        seq_put_decimal_ull(m, ' ', mm->start_data);
        seq_put_decimal_ull(m, ' ', mm->end_data);
        seq_put_decimal_ull(m, ' ', mm->start_brk);
        seq_put_decimal_ull(m, ' ', mm->arg_start);
        seq_put_decimal_ull(m, ' ', mm->arg_end);
        seq_put_decimal_ull(m, ' ', mm->env_start);
        seq_put_decimal_ull(m, ' ', mm->env_end);
        seq_put_decimal_ll(m, ' ', task->exit_code)
```

those may what we need to add.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

(Reply via y...@solvcon.net):

Reasonable.  I am wondering that why Ubuntu 12.10 enables those fields, and
would they later disable them.  Anyway, at the time being, it's good to
have a patch to fix this compatibility issue.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

It is resolved in commit 7c52637.  Thanks to Robert.
